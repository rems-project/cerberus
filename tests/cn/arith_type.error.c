/*@

function (integer) negate() {
 -5
}

function (integer) not_bool() {
 let x = negate() + true;
 !x
}


@*/

