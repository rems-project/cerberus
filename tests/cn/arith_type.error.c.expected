return code: 1
tests/cn/arith_type.error.c:8:21: error: Type error
 let x = negate() + true;
                    ^
Expression 'true' has type 'boolean'.
I expected it to have type 'integer' because of tests/cn/arith_type.error.c:8:10:
 let x = negate() + true;
         ~~~~~~^~ 
